### PR TITLE
Improve error handling and add some logging

### PR DESCRIPTION
**Pull request by JACooper**

### CODE REVIEW

**Comment by JACooper** [comment]
[Investigate issues with image upload into notes failing mysteriously](https://trello.com/c/8xlrLN5b/3699-investigate-issues-with-image-upload-into-notes-failing-mysteriously)

File: src/plugin.js [hunk 1]
```diff
@@ -2,6 +2,9 @@ function initPlugin() {
   const onFileInput = function(editor, file) {
     // We use TinyMCE's editor.dom here (rather than document.createElement) as it is empirically more reliable within the editor
     const img = editor.dom.create('img', {id: `tinymce-new-image-${file.name}`});
+    if (!img && honeybadger) {
```
Comment (antifun) [inline src/plugin.js]: Lowercase honeybadger?

File: src/plugin.js [hunk 2]
```diff
@@ -2,6 +2,9 @@ function initPlugin() {
...
+      honeybadger.notify({ desc: 'Could not create img tag', fileObj: file });
```
[the remark this answers is not in the document]
Comment (antifun) [reply src/plugin.js]: Perhaps log everything but the raw image, then?

File: src/plugin.js [hunk 3]
```diff
@@ -25,10 +31,20 @@ function initPlugin() {
...
         (err) => {
-          alert('Upload failed');
+          editor.dom.remove(`*[id="tinymce-new-image-${file.name}"]`);
+          editor.getParam('image_conversion_error_handler')();
         }
       );
     }
+
+    reader.onerror = () => {
+      reader.abort()
```
Comment (antifun) [inline src/plugin.js]: Might as well log to the console too, if only to give users a chance to help us find these. WDYT?

File: src/plugin.js [hunk 4]
```diff
@@ -10,6 +13,9 @@ function initPlugin() {
     reader.onload = () => {
       const base64Data = reader.result;
```
Comment (antifun) [inline src/plugin.js]: Are we sure this always has some value in it?

File: src/plugin.js (hunk 1 above)
Comment (JACooper) [reply src/plugin.js]: Ah. . .good catch. Should be a capital h.

File: src/plugin.js (hunk 4 above)
Comment (JACooper) [reply src/plugin.js]: From what I can tell based on the `onload` literature, yes.

File: src/plugin.js (hunk 3 above)
Comment (JACooper) [reply src/plugin.js]: Seems reasonable.

File: src/plugin.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (JACooper) [reply src/plugin.js]: :+1:

[the remark this answers is not in the document]
Comment (JACooper) [reply src/plugin.js]: Actually, I'm totally wrong. Logging `fileObj` doesn't seem to put any raw blob data in.

**Pull request merged**